usr/lib/common/key.c

```diff
@@ -1910,44 +2253,118 @@ CK_RV rsa_priv_unwrap(TEMPLATE *tmpl, CK_BYTE *data, CK_ULONG total_length)
-    template_update_attribute(tmpl, exponent1);
-    template_update_attribute(tmpl, exponent2);
-    template_update_attribute(tmpl, coeff);
-
-    return CKR_OK;
-}
-
-
-CK_RV rsa_priv_unwrap_get_data(TEMPLATE *tmpl,
-                              CK_BYTE *data, CK_ULONG total_length)
-{
-    CK_ATTRIBUTE *modulus = NULL;
-    CK_ATTRIBUTE *publ_exp = NULL;
-    CK_RV rc;
-
-    rc = ber_decode_RSAPublicKey(data, total_length, &modulus, &publ_exp);
-
+    rc = template_update_attribute(tmpl, modulus);
     if (rc != CKR_OK) {
-        TRACE_DEVEL("ber_decode_RSAPublicKey failed\n");
-        return rc;
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    modulus = NULL;
+    rc = template_update_attribute(tmpl, publ_exp);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    publ_exp = NULL;
+    rc = template_update_attribute(tmpl, priv_exp);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    priv_exp = NULL;
+    rc = template_update_attribute(tmpl, prime1);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    prime1 = NULL;
+    rc = template_update_attribute(tmpl, prime2);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    prime2 = NULL;
+    rc = template_update_attribute(tmpl, exponent1);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    exponent1 = NULL;
+    rc = template_update_attribute(tmpl, exponent2);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    exponent2 = NULL;
+    rc = template_update_attribute(tmpl, coeff);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("template_update_attribute failed.\n");
+        goto error;
+    }
+    coeff = NULL;
+
+    return CKR_OK;
+
+error:
+    if (modulus)
+        free(modulus);
+    if (publ_exp)
+        free(publ_exp);
+    if (priv_exp)
+        free(priv_exp);
+    if (prime1)
+        free(prime1);
+    if (prime2)
+        free(prime2);
+    if (exponent1)
+        free(exponent1);
+    if (exponent2)
+        free(exponent2);
+    if (coeff)
+        free(coeff);
+
+    return rc;
+}
+
+
+CK_RV rsa_priv_unwrap_get_data(TEMPLATE *tmpl,
+                              CK_BYTE *data, CK_ULONG total_length)
+{
+    CK_ATTRIBUTE *modulus = NULL;
+    CK_ATTRIBUTE *publ_exp = NULL;
+    CK_RV rc;
+
+    rc = ber_decode_RSAPublicKey(data, total_length, &modulus, &publ_exp);
+
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("ber_decode_RSAPublicKey failed\n");
+        return rc;
     }
 
-    p11_attribute_trim(modulus);
-    p11_attribute_trim(publ_exp);
-
+    p11_attribute_trim(modulus);
+    p11_attribute_trim(publ_exp);
```

Comment:
Just out of curiosity: What is the diff here?  Whitespace?  Looks the same to me...

Reply:
hm, in the current version on github this lines don't show any difference.....